src/util/loanSearch/queryParamUtils.js

```diff
@@ -23,12 +40,25 @@ const lendToFlssSort = new Map([
 export function getIdsFromQueryParam(param, names, facets) {
 	if (!param) return;
 
-	// Handles FLSS and legacy query params, such as "1" and "1,2"
+	// Handles FLSS and legacy query params, such as "1" and "1,2" AND chained theme names
 	if (param.includes(',') || isNumber(param)) {
-		return param.split(',').filter(s => s !== '').map(s => +s);
+		return param.split(',').reduce((prev, current) => {
+			const name = current.toUpperCase();
+			if (names.includes(name)) {
+				const facet = facets.find(s => s.name.toUpperCase() === name);
+
+				if (facet) {
+					prev.push(facet.id);
+				}
+			} else if (current !== '') {
+				return [...prev, parseInt(current, 10)];
+			}
+
+			return prev;
+		}, []);
```

Comment:
@dyersituations this is certainly uglier than your original process for comma separated values but enabled us to support lists of theme names.

Reply:
When are there lists of theme names? I thought it was comma separated IDs or names with ~

Reply:
There was at least one category in the query map that had 2 theme names so this was really to support that. A single name already worked with the small addition below!

Reply:
Gotcha. I wonder if that was an accident. No worries supporting a different use case though!

Reply:
Yea, a single theme name would go into the final process and get matched to the facet! Figured if it worked there, I could use for the comma separated process too...

I suppose in the future we could actually extract the query value prep and use a single reduce process for all formats.